src/network/adapter.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#include "adapter.hpp"
+
+#include "dbus.hpp"
+
+#include <phosphor-logging/log.hpp>
+
+#include <charconv>
+#include <fstream>
+
+using namespace phosphor::logging;
+
+static constexpr const char* pciidsPath = "/usr/share/misc/pci.ids";
+
+const std::string NetworkAdapter::inventoryPath =
+    dbus::inventory::pathBase + std::string("/system/network/adapter/");
+
+const std::pair<std::string, std::string>
+    NetworkAdapter::pciLookup(const std::string& vendorId,
+                              const std::string& deviceId)
+{
+    static const size_t vidPrefixLen = 0;
+    /* The line of DID has one tabulation to indent */
+    static const size_t didPrefixLen = 1;
+    static const size_t wordLen = 4;
+    static const size_t sepLen = 2;
+    /* The 0x prefix of the VID: `0x____` */
+    static const size_t hexPrefixLen = 2; 
+
+    std::string line;
+    std::string vendorName;
+    std::string modelName;
+    std::ifstream idfile(pciidsPath);
+    if (!idfile.is_open())
+    {
+        log<level::ERR>("failed to open pid.ids file");
+        return std::pair<std::string, std::string>();
+    }
+
+    while (std::getline(idfile, line))
+    {
+        std::string curVIDstr = line.substr(vidPrefixLen, wordLen);
+        if (vendorName.empty() &&
+            (curVIDstr.compare(vendorId.substr(hexPrefixLen)) == 0))
```

Comment:
I dislike the `substr` using as a temporary value because it leads to an excess allocation for this temporary string.
You could use the other overload of `compare` with a position as an argument.
https://en.cppreference.com/w/cpp/string/basic_string/compare

Reply:
fixed